### PR TITLE
feat: Add linting

### PR DESCRIPTION
**Pull request by wyrmisis**
## Intent
This is step one of a few PRs I hope to put up around linting. This one...
* Adds dependencies for linting and formatting JS/TS
* Configures linting tools
* Prepares us to fix linting issues, manually and automatically (with `npm run fix`)
* Runs linting with `npm run lint`, with a fairly strict ruleset. We can pare this back if need be.

### Next step PRs
* **Fixing our documentation- and typing-related linting errors**: This should wait until we're 100% through the data model changes, as a fair number of class members will be removed/moved around.
* **Fixing our logical/code linting errors:** This will be a fairly big chunk of work with manual review necessary.
* **Blocking non-compliant commits:** This will add complexity to the repo, as new devs will be expected to conform to our coding style; we should hold off on this until we've eliminated all of our existing errors.

### CODE REVIEW

**Review by anthonyronda — COMMENTED** [review]
I am not ready to put this in 1.7.0 due to the higher-than-normal changeset we're delivering with the data model changes. This may be a bit much for module devs and casual contributors. After 1.7.0 is in, I'll find the time to reduce the number of errors and warnings, and then we'll be ready to put this in main

File: rollup.config.js [hunk 1]
```diff
@@ -35,6 +36,9 @@ export default defineConfig([
     plugins: [
       nodeResolve(),
       typescript(),
+      eslint({
+        fix: !isWatchMode
```
[the remark this answers is not in the document]
Comment (wyrmisis) [reply rollup.config.js:40]: This fixes as part of the build. We could probably remove this.

File: .browserslistrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+defaults and supports es6-module and supports array-find and supports array-flat and supports array-includes
+Electron > 20
```
[the remark this answers is not in the document]
Comment (wyrmisis) [reply .browserslistrc:2]: I thought the Electron update would've come along with a v10-friendly client. Huh, weird.

[the remark this answers is not in the document]
Comment (anthonyronda) [reply .browserslistrc:2]: Electron updates require full uninstall-reinstalls so Foundry Gaming tries to limit requiring them to only when 100% necessary

[the remark this answers is not in the document]
Comment (anthonyronda) [reply .browserslistrc:2]: I believe Electron > 20 is the appropriate setting here regardless

**Review by anthonyronda — APPROVED** [review]
Makes sense, let's get started! :D

**Pull request merged**